tools/ci/compile_ci.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+#
+# This script is the CI entrypoint for compiling Flink and running QA checks that don't require tests.
+#
```

Comment:
```suggestion
if [[ ! -z "${AZURE_LOCATION}" ]]; then
  echo 'No "$AZURE_LOCATION" environmental found. This script is the CI entrypoint for compiling Flink and running QA checks that don't require tests. Please use ./compile.sh for local setups.'
  exit 1
fi
```

Reply:
IMO we shouldn't put additional roadblocks into the way if you want to run the CI version locally. It will work (I guess).